### PR TITLE
Memory pool support for Replication IO

### PR DESCRIPTION
**Pull request by somnathr**
Added Memory pool support for Replication IO.

## Description

Memory pool support for Replication IO. Configurable three different memory pool size and based on the IO size the closest memory pool will be selected. This will enable IO size > 1MB in the DSS Minio KV stack.
## Motivation and Context

To support > 1MB IO in the Minio KV path.
## How Has This Been Tested?

Unit testing with single node setup. Performance tested on single minio with 1MB and 8 MB.
**Disclaimer** because of target issue of supporting large number of bigger IOs, can't stress with more ios in my test.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/erasure-utils.go [hunk 1]
```diff
@@ -73,6 +73,125 @@ func (k *kvMetaPoolECPoolType) PrintCount() {
...
+                       b := make([]byte, midSize)
+                       return b
+               },
+        },
+  }
+
+  divFactor = 2 * divFactor
+  minSize = globalMaxKVObject/int64 (divFactor)
+  for (minSize <= 32768) {
+    midSize = (2 * globalMaxKVObject)/int64 (divFactor)
```
Comment (rajpayagond) [inline cmd/erasure-utils.go]: allocating for minSize not midSize

**Review by rajpayagond — CHANGES_REQUESTED** [review]
Please update the minSize allocation

File: cmd/erasure-utils.go (hunk 1 above)
Comment (somnathr) [reply cmd/erasure-utils.go]: good catch, will resolve, thx

**Review by rajpayagond — APPROVED** [review]

**Review by rajpayagond — APPROVED** [review]
Looks good.

**Comment by rajpayagond** [comment]
Done

**Review by payagond — APPROVED** [review]
Looks good!

**Pull request merged**